### DISCUSSION TITLE
Include pull request metadata in merge_queue's checks_requested webhook payload

### DISCUSSION THREAD

**Discussion by janeklb** [community/community#62219] [Apps, API and Webhooks]
### Select Topic Area

Product Feedback

### Body

My merge queue workflow would benefit if the webhook payload included explicit metadata about the pull request being merged (as part of the merge queue).

In my scenario, I need the `pull_number` and PR's head ref. This data is indirectly available by looking at the payload's `head_ref` value to extract the `pull_number` and then making a request to the github API to get the PR's head ref.

This is undesirable because the format of head ref (currently something like `refs/heads/gh-readonly-queue/main/pr-1969-e4bb86df2887f0bd3e04e3b4e9dc15075a83c13c`) is not part of the webhook API contract, which makes it a brittle thing to rely on.

Suggestion:
Add some explicit PR metadata to the webhook payload (eg. `pr_head_ref`)

ps:
What i'm _ultimately_ after is the `merge-base` of my PR's HEAD and the PR's target branch (eg. `git merge-base <PRs-head-ref> <PRs-target>`). If that could be included in the webhook payload, that would be even better!

**Comment by Emilgardis** [comment]
I want this also, specifically to check what PRs are included and grab metadata from them to determine what actions should be ran (or not) via labels.

**Comment by joehorsnell** [comment]
Linking to https://github.com/orgs/community/discussions/65239 for visibility, which has more upvotes
